lib/fs.js

```diff
@@ -42,6 +42,14 @@ function throwOptionsError(options) {
     'but got ' + typeof options + ' instead');
 }
 
+function sanitizeFD(inputFD) {
+  const fd = +inputFD;
+  if (!Number.isInteger(fd) || fd < 0 || fd > 0xFFFFFFFF) {
```

Comment:
You're coercing to a double, then range checking. Meaning floating point values are getting through. If you want to get around this do:
```
function sanitizedFD(inputFD) {
  if (fd < 0 || fd > 0xFFFFFFFF || fd >>> 0 !== fd)
    throw new TypeError('file descriptor must be a signed 32-bit integer');
```